fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java

```diff
@@ -1386,6 +1463,16 @@ private void computeColocatedJoinInstanceParam(PlanFragmentId fragmentId, int pa
                 params.instanceExecParams.add(instanceParam);
             }
         }
+        boolean dopAdaptionEnabled = ConnectContext.get() != null &&
+                ConnectContext.get().getSessionVariable().isPipelineDopAdaptionEnabled();
+        // ensure numInstances * pipelineDop = degreeOfParallelism when dop adaptation is enabled
+        if (dopAdaptionEnabled) {
+            int numInstances = params.instanceExecParams.size();
+            int numBackends = addressToScanRanges.size();
+            int degreeOfParallelism = ConnectContext.get().getSessionVariable().getDegreeOfParallelism();
+            int pipelineDop = Math.max(1, degreeOfParallelism / Math.max(1, numInstances / numBackends));
```

Comment:
If dopAdaptionEnabled is true,  parallelExecNum is set to half the average number of hardware cores of backends, and pipelineDop is set to 1. why we need to change pipelineDop here

Reply:
for Pipelines with ScanOperator as its source operator, pipelineDop determines its the number of concurrent io tasks, if pipeineDop is too small, then io operations delay the processing of the Pipeline.